ehrapy/tools/feature_ranking/_feature_importances.py

```diff
@@ -0,0 +1,163 @@
+        >>> ep.pp.knn_impute(adata, n_neighbours=5)
+        >>> input_features = [
+        ...     feat for feat in adata.var_names if feat not in {"service_unit", "day_icu_intime", "tco2_first"}
+        ... ]
+        >>> ep.tl.rank_features_supervised(adata, "tco2_first", "continuous", "rf", input_features=input_features)
+    """
+    if predicted_feature not in adata.var_names:
+        raise ValueError(f"Feature {predicted_feature} not found in adata.var.")
+
+    if input_features != "all":
+        for feature in input_features:
+            if feature not in adata.var_names:
+                raise ValueError(f"Feature {feature} not found in adata.var.")
+
+    if model not in ["regression", "svm", "rf"]:
+        raise ValueError(f"Model {model} not recognized. Please choose either 'regression', 'svm', or 'rf'.")
+
+    if prediction_type not in ["continuous", "categorical"]:
+        raise ValueError(
+            f"Prediction type {prediction_type} not recognized. Please choose either 'continuous' or 'categorical'."
+        )
+
+    if feature_scaling not in ["standard", "minmax", None]:
+        raise ValueError(
+            f"Feature scaling type {feature_scaling} not recognized. Please choose either 'standard', 'minmax', or None."
+        )
+
+    data = anndata_to_df(adata, layer=layer)
+
+    if prediction_type == "continuous":
+        if pd.api.types.is_categorical_dtype(data[predicted_feature].dtype):
+            try:
+                data[predicted_feature] = data[predicted_feature].astype(float)
+            except ValueError as e:
+                raise ValueError(
+                    f"Feature {predicted_feature} is not continuous and conversion to float failed. Either change the prediction "
+                    f"type to 'categorical' or change the feature data type to a continuous type."
+                ) from e
+
+        if model == "regression":
+            predictor = LinearRegression(**kwargs)
+        elif model == "svm":
+            predictor = SVR(kernel="linear", **kwargs)
+        elif model == "rf":
+            predictor = RandomForestRegressor(**kwargs)
+
+    elif prediction_type == "categorical":
+        if not pd.api.types.is_categorical_dtype(data[predicted_feature].dtype):
+            try:
+                data[predicted_feature] = data[predicted_feature].astype("category")
+            except ValueError as e:
+                raise ValueError(
+                    f"Feature {predicted_feature} is not categorical and conversion to category failed. Either change the prediction "
+                    f"type to 'continuous' or change the feature data type to a categorical type."
+                ) from e
+
+        if data[predicted_feature].nunique() > 2 and model in ["regression", "svm"]:
+            raise ValueError(
+                f"Feature {predicted_feature} has more than two categories. Please choose random forest (rf) as model for multi-class classification."
```

Comment:
```suggestion
                f"Feature {predicted_feature} has more than two categories. Please choose 'rf' as model for multi-class classification."
```

Maybe it's simpler like this?